Apps/CocoaPods-FCM/src/View/SettingsView.swift

```diff
@@ -70,7 +78,48 @@ struct SettingsView: View {
                     viewModel.settings.apiKey = apiKey
                 }
             }
+            .alert(isPresented: .notNil(alertMessage)) {
+                Alert(
+                    title: Text("Error"),
+                    message: Text(alertMessage!),
+                    dismissButton: .default(Text("OK")) {
+                        alertMessage = nil
+                    }
+                )
+            }
+        }
+    }
+
+    private func verifyTrackUrl() -> Bool {
+        let enteredUrl = viewModel.settings.trackUrl
+
+        guard !enteredUrl.isEmpty else {
+            alertMessage = "Tracking URL is empty. Therefore, I cannot save the settings."
+            return false
         }
+
+        guard let url = URL(string: enteredUrl) else {
```

Comment:
how about moving all validations to a separate function to make the code look clean and modulate. This will also help us set up test cases easily incase required to do in the future.

Reply:
the validations are all in a separate function, verifyTrackUrl().

Reply:
Ah ! Yes... my bad ! 🤣